data_diff/diff_tables.py

```diff
@@ -403,6 +404,15 @@ def _diff_tables(self, table1, table2, level=0, segment_index=None, segment_coun
             f"size: {table2.max_key-table1.min_key}"
         )
 
+        # When benchmarking, we want the ability to download the segments in
+        # parallel. By default, data-diff will checksum the section first (when
+        # it's below the threshold) and _then_ download it.
+        if BENCHMARK:
```

Comment:
FWIW I don't think it would be crazy to make this the default... 

I ran the benchmark with the default bisection threshold, and checksumming / downloading with the default threshold and ~10k rows is extremely competitive. And the current approach is twice as slow when there is a difference.